src/app/node-data/component.ts

```diff
@@ -191,35 +191,44 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
   }
 
   isOperatingSystemSupported(os: OperatingSystem): boolean {
-    // If VSphere is selected enable OS only if it is also defined in the datacenter spec
-    if (this._hasSystemTemplate(NodeProvider.VSPHERE, os)) {
-      return true;
-    }
+    if (this._datacenterSpec.spec.enabledOperatingSystems.length > 0) {
```

Comment:
`enabledOperatingSystems` is optional, if it will be undefined then the line above will produce errors.